drivers/video/hm01b0.c

```diff
@@ -261,10 +441,84 @@ static bool hm01b0_check_connection(const struct device *dev)
 	return (model_id == HM01B0_ID);
 }
 
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(pwdn_gpios) || DT_ANY_INST_HAS_PROP_STATUS_OKAY(reset_gpios)
+static bool hm01b0_try_reset_pwdn_pins(const struct device *dev, uint8_t iter)
+{
+	const struct hm01b0_config *config = dev->config;
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(pwdn_gpios)
+	gpio_pin_set_dt(&config->pwdn, iter >> 1);
+#else
+	if (iter >> 1) {
+		return false; /* cut iterations in half if not defined */
+	}
+#endif
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(reset_gpios)
+	gpio_pin_set_dt(&config->reset, iter & 1);
+#else
+	if (iter & 1) {
+		return false; /* cut iterations in half if not defined */
+	}
+#endif
+	/* lets try a couple of iterations before we punt */
+	uint8_t retry_count = 3;
+
+	while (retry_count) {
+		k_sleep(K_MSEC(10));
+		if (hm01b0_check_connection(dev)) {
+			LOG_INF("Reset/pwdn pins valid:%u", retry_count);
+			return true;
+		}
+		retry_count--;
+	}
+	return false;
+}
+#endif
+
 static int hm01b0_init(const struct device *dev)
 {
+	const struct hm01b0_config __maybe_unused *config = dev->config;
 	int ret;
 
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(pwdn_gpios) || DT_ANY_INST_HAS_PROP_STATUS_OKAY(reset_gpios)
```

Comment:
You do not need that "double-`#ifdef`", or at least not at this place, proposition to move it below to simplify a bit.